### ISSUE TITLE
Use an (static) adaptive mesh after it is saved and loaded

### ISSUE THREAD

**Issue by tangqi** [mfem/mfem#1326]
Hi,
I have some issues to use the adaptive mesh after it is saved.
![Screen Shot 2020-02-27 at 5 26 39 PM](https://user-images.githubusercontent.com/1406660/75499141-613b7e00-5986-11ea-8016-b171ccbf3fe9.png)

This mesh is obtained by a simple local refinement through mesh-explorer.cpp. (It is periodic in one direction). 

Now if I do a simple `ProjectCoefficient` to H1 space, this is what I got:
![Screen Shot 2020-02-27 at 5 30 29 PM](https://user-images.githubusercontent.com/1406660/75499355-f9d1fe00-5986-11ea-92a1-152a8c0179de.png)
But the function should be 
![Screen Shot 2020-02-27 at 5 31 13 PM](https://user-images.githubusercontent.com/1406660/75499407-1b32ea00-5987-11ea-8422-e3697d035cea.png)

Do I miss something when I loaded this mesh? The test code is very simple
```c++
   Mesh *mesh = new Mesh(mesh_file, 1, 1);
   ParMesh *pmesh = new ParMesh(MPI_COMM_WORLD, *mesh);
...
   ParGridFunction psi(&fespace);
   FunctionCoefficient psiInit(func);
   psi.ProjectCoefficient(psiInit);
```
 I think if I do not save this mesh and instead generate it on the fly, it seems okay with projections. But it would nice if I can directly use a mesh like this. Thanks!

**Comment by jakubcerveny** [comment]
Hi @tangqi, could you please attach the mesh that you created? Thanks

**Comment by tangqi** [comment]
Thanks Jakub, @jakubcerveny

See mesh:
https://github.com/mfem/mfem/blob/tds-mhd-dev/miniapps/mhd/Meshes/amr-mesh-new.mesh

It is created by mesh-explorer by a simple local refinement based on the coarse mesh:
https://github.com/mfem/mfem/blob/tds-mhd-dev/miniapps/mhd/Meshes/xperiodic-new.mesh

Here is a simple serial test code:
```c++
#include "mfem.hpp"
#include <memory>
#include <iostream>
#include <fstream>

using namespace std;
using namespace mfem;

double InitialPsi(const Vector &x)
{
   double beta=1e-3, lambda=.5/M_PI, ep=.2;
   return -lambda*log( cosh(x(1)/lambda) +ep*cos(x(0)/lambda) )
          +beta*cos(M_PI*.5*x(1))*cos(M_PI*x(0));
}

int main(int argc, char *argv[])
{
   //++++Parse command-line options.
   const char *mesh_file = "./amr-periodic.mesh";
   int ser_ref_levels = 2;
   int order = 2;

   OptionsParser args(argc, argv);
   args.AddOption(&mesh_file, "-m", "--mesh",
                  "Mesh file to use.");
   args.AddOption(&ser_ref_levels, "-rs", "--refine",
                  "Number of times to refine the mesh uniformly in serial.");
   args.AddOption(&order, "-o", "--order",
                  "Order (degree) of the finite elements.");

   args.Parse();
   if (!args.Good())
   {
      args.PrintUsage(cout);
      return 1;
   }

   //+++++Read the mesh from the given mesh file.    
   Mesh *mesh = new Mesh(mesh_file, 1, 1);
   int dim = mesh->Dimension();

   //++++++Refine the mesh to increase the resolution.    
   mesh->EnsureNCMesh();
   for (int lev = 0; lev < ser_ref_levels; lev++)
   {
      mesh->UniformRefinement();
   }

   H1_FECollection fe_coll(order, dim);
   FiniteElementSpace fespace(mesh, &fe_coll); 

   GridFunction psi(&fespace);
   FunctionCoefficient psiInit(InitialPsi);
   psi.ProjectCoefficient(psiInit);
   psi.SetTrueVector();
   psi.SetFromTrueVector(); 

   socketstream vis_phi;
   char vishost[] = "localhost";
   int  visport   = 19916;
   vis_phi.open(vishost, visport);

   vis_phi.precision(8);
   vis_phi << "solution\n" << *mesh << psi;
   vis_phi << "window_size 800 800\n"<< "window_title '" << "psi'" << "keys cm\n";
   vis_phi << flush;

   delete mesh;
   return 0;
}
```

In serial, it appears to work with no refinement (rs=0) but it becomes bad when refineing. But in parallel, it gives me bad projected solutions no matter what. The parallel testing code can be found at https://github.com/mfem/mfem/blob/tds-mhd-dev/miniapps/mhd/old/testAMR.cpp

It was fine inside mesh-explorer if we do projections or further refinement.

**Comment by tangqi** [comment]
OK, I work around by generating this mesh and partitioning on the fly. (Another issue seems to be `ParMesh(MPI_COMM_WORLD, *mesh)` will get confused by `ncmesh`, so I need to generate partitioning using serial meshes first.) 

It is not critical to fix it at this point. Thanks again for the help. -qi

**Comment by jakubcerveny** [comment]
I will look into this together with #1332

**Comment by tangqi** [comment]
I can confirm the new nonconforming mesh v1.0 fixed this issue
